packages/server/utils/makeCreateJiraTaskComment.ts

```diff
@@ -0,0 +1,71 @@
+import {Doc} from './convertContentStateToADF'
+import {ExternalLinks} from '../../client/types/constEnums'
+
+const makeCreateJiraTaskComment = (
+  creator: string,
+  assignee: string,
+  teamName: string,
+  teamDashboardUrl: string
+): Doc => ({
```

Comment:
nit: We try to avoid specifying return type as it's easier to maintain in a long term. TS will infer the type when the return type has a type specified, for example
```
const makeCreateJiraTaskComment = (
  creator: string,
  assignee: string,
  teamName: string,
  teamDashboardUrl: string
) =>
  ({
    version: 1 as const,
    type: 'doc' as const,
    content: [
      {
        type: 'paragraph',
        content: [
          {
            type: 'text',
            text: `Created by ${creator} for ${assignee}`
          },
          {
            type: 'hardBreak'
          },
          {
            type: 'text',
            text: 'See the dashboard of '
          },
          {
            type: 'text',
            text: teamName,
            marks: [
              {
                type: 'link',
                attrs: {
                  href: teamDashboardUrl
                }
              }
            ]
          },
          {
            type: 'hardBreak'
          },
          {
            type: 'text',
            text: 'Powered by ',
            marks: [
              {
                type: 'em'
              }
            ]
          },
          {
            type: 'text',
            text: 'Parabol',
            marks: [
              {
                type: 'link',
                attrs: {
                  href: ExternalLinks.GETTING_STARTED_CHECK_INS
                }
              },
              {
                type: 'em'
              }
            ]
          }
        ]
      }
    ]
  } as Doc)
```

Reply:
> Great job, all the scenarios work as expected. Only a couple of minor comments, after that let's move it to maintainer review.
> 
> One question, there's a small difference in the way we handle Jira vs Github integration. ie. Jira has this condition:
> 
> ```
> if (!userId) {
>       // we should probably remove this constraint in the future
>       return {error: {message: 'Task must have assignee before it can be integrated with Jira'}}
>     }
> ```
> 
> any reason to keep it?

Probably not. It was added [here](https://github.com/ParabolInc/parabol/pull/4312/files#diff-13ccf9270fcf560f88b8b298fced28727d4aa6e20731d7537f7d18d39ad8d896) without any explanation in PR or commit message. @mattkrick can this be removed?

Reply:
oh, nice! I didn't know that. 👌